lasy/utils/laser_utils.py

```diff
@@ -873,3 +873,229 @@ def import_from_z(dim, grid, omega0, field_z, z_axis, z0=0.0, t0=0.0, backend="N
+
+    Return
+    ----------
+    sigma: Standard deviation of a**2 in m
+    """
+    field = grid.get_temporal_field()
+    if dim == "xyt":
+        Nx, Ny, Nt = field.shape
+        A2 = (np.abs(field[Nx // 2 - 1, :, :]) ** 2).sum(-1)
+        ax = grid.axes[1]
+    else:
+        A2 = (np.abs(field[0, :, :]) ** 2).sum(-1)
+        ax = grid.axes[0]
+        if ax[0] > 0:
+            A2 = np.r_[A2[::-1], A2]
+            ax = np.r_[-ax[::-1], ax]
+        else:
+            A2 = np.r_[A2[::-1][:-1], A2]
+            ax = np.r_[-ax[::-1][:-1], ax]
+
+    sigma = 2 * np.sqrt(np.average(ax**2, weights=A2))
+
+    return sigma
+
+
+def get_phi2(dim, grid):
+    r"""
+    Calculate the Group-delay dispersion of the laser.
+
+    Parameters
+    ----------
+    dim : string
+        Dimensionality of the array. Options are:
+        - 'xyt': The laser pulse is represented on a 3D grid:
+                 Cartesian (x,y) transversely, and temporal (t) longitudinally.
+        - 'rt' : The laser pulse is represented on a 2D grid:
+                 Cylindrical (r) transversely, and temporal (t) longitudinally.
+
+    grid : a Grid object.
+        It contains an ndarray (V/m) with the value of the envelope field and the associated metadata that defines the points at which the laser is defined.
+
+     Return
+    ----------
+    phi2: Group-delayed dispersion in :math:`\Phi^{(2)}=d(\omega_0)/dt` (second^-2)
+    varphi2: Group-delayed dispersion in :math:`\varphi^{(2)}=dt_0/d(\omega)` (second^2)
+    """
+    tau = 2 * get_duration(grid, dim)
+    env = grid.get_temporal_field()
+    env_abs2 = np.abs(env**2)
+    # Calculate group-delayed dispersion
+    phi_envelop = np.unwrap(np.array(np.arctan2(env.imag, env.real)), axis=2)
+    pphi_pt = np.gradient(phi_envelop, grid.dx[-1], axis=2)
+    pphi_pt2 = np.gradient(pphi_pt, grid.dx[-1], axis=2)
+    phi2 = np.average(pphi_pt2, weights=env_abs2)
+    varphi2 = np.max(np.roots([4 * phi2, -4, tau**4 * phi2]))
+    return phi2, varphi2
+
+
+def get_zeta(dim, grid, k0):
+    r"""
+    Calculate the spatio-chirp of the laser.
+
+    Parameters
+    ----------
+    dim : string
+        Dimensionality of the array. Options are:
+        - 'xyt': The laser pulse is represented on a 3D grid:
+                 Cartesian (x,y) transversely, and temporal (t) longitudinally.
+        - 'rt' : The laser pulse is represented on a 2D grid:
+                 Cylindrical (r) transversely, and temporal (t) longitudinally.
+
+    grid : a Grid object.
+        It contains an ndarray (V/m) with the value of the envelope field and the associated metadata that defines the points at which the laser is defined.
+
+     Return
+    ----------
+    zeta_x, zeta_y: Spatio-chirp in :math:`\zeta=dx_0/d(\omega_0)` (meter * second)
+    nu_x, nu_y: Spatio-chirp in :math:`\nu=d(\omega_0)/dx` (meter^-1 * second^-1)
```

Comment:
Also, please fix space before colon for parameters everywhere!
```suggestion
    nu_x, nu_y : Spatial chirp in :math:`\nu=d(\omega_0)/dx` (meter^-1 * second^-1)
```